src/main/java/brave/opentracing/BraveActiveSpanSource.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2016-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.opentracing;
+
+import brave.Tracer;
+import brave.Tracer.SpanInScope;
+import brave.Tracing;
+import io.opentracing.ActiveSpan;
+import io.opentracing.ActiveSpanSource;
+import io.opentracing.Span;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Manages the active Brave span.
+ *
+ * Note that it is important to use the OpenTracing API exclusively to activate spans. If you fail
+ * to do this, querying for the active span using the OpenTracing API will throw an exception.
+ */
+final class BraveActiveSpanSource implements ActiveSpanSource {
+  private final Map<Long, BraveActiveSpan> activeSpans = new ConcurrentHashMap<>();
```

Comment:
The returned reference of the span in scope is usually a thread-scoped reference, but not always. Attempting to store a reference widens the implementation contract of brave (and census, and guice, and similar's) more than needed, because the safe usage pattern is always closing it on the same thread you opened it on.

For what you are trying to do here, it seems you are actually constrained to a thread-local implementation. For that, the ThreadLocal itself is a tool you could use. For example, you have a ThreadLocal here in this code, which catches the SpanInScope. you can check that thread local later.